### PR TITLE
Add avif conversion v1

### PR DESCRIPTION
**Pull request by AndhikaRei**
### Changes
- Add AVIF conversion
- Upgraded to Imagick V7+++
- Modify and change deprecated function to newer one
- Add optimized blur function
- Add sample nginx.conf and images for testing
- Add Docker support for easier installation in local
- Add more complete README.MD, especially for installation
- Add UPGRADING.MD to list changes needed on ngx_small_light and nginx

### Issues
- None

### Notes
- 

### Testing scenario
- Run the application using docker by typing "docker-compose up -d"
- You can test the endpoint using Postman. You can download test collection [here](https://drive.google.com/drive/folders/1tRmIMcBH-t6ozGi2epwUMPEjXjWuHInS?usp=sharing) 

### Expected Result / Output
- Able to convert various image type to AVIF
- Able to convert AVIF to various image type
- Previous functionality of ngx_small_light is still working
- Able to use optimized blur function together with other functionality

### Reviewer
- [ ] @midorinet

### CODE REVIEW

File: src/ngx_http_small_light_imagemagick.c [hunk 1]
```diff
@@ -87,77 +90,76 @@ void ngx_http_small_light_imagemagick_term(void *data)
...
+) {
     ngx_http_small_light_imagemagick_ctx_t *ictx;
     ngx_http_small_light_image_size_t       sz;
     MagickBooleanType                       status;
-    int                                     rmprof_flg, progressive_flg, cmyk2rgb_flg;
+    int                                     rmprof_flg, progressive_flg, cmyk2rgb_flg, bluropt_flag;
     double                                  iw, ih, q;
     char                                   *unsharp, *sharpen, *blur, *of, *of_orig;
-    MagickWand                             *trans_wand, *canvas_wand, *canvas_bg_wand, *source_wand;
+    MagickWand                             *canvas_wand, *canvas_bg_wand, *source_wand;
```
Comment (kawakibi) [inline src/ngx_http_small_light_imagemagick.c]: `*canvas_bg_wand` bisa diapus karena di block code 

```
// check background fill
...
```

gak kita pakai lagi

File: src/ngx_http_small_light_imagemagick.c [hunk 2]
```diff
@@ -87,77 +90,76 @@ void ngx_http_small_light_imagemagick_term(void *data)
...
-    MagickWand                             *icon_wand;
+    MagickWand                             *icon_wand, *merge_wand;
     u_char                                 *p, *embedicon;
     size_t                                  embedicon_path_len, embedicon_len, sled_image_size;
     ngx_int_t                               type;
-    u_char                                  jpeg_size_opt[32], crop_geo[128], size_geo[128], embedicon_path[256];
+    u_char                                  jpeg_size_opt[32], embedicon_path[256];
     ColorspaceType                          color_space;
 #if MagickLibVersion >= 0x690
     int                                     autoorient_flg, backgroundfill_flg;
```
Comment (kawakibi) [inline src/ngx_http_small_light_imagemagick.c]: `backgroundfill_flg` juga bisa dihapus

File: src/ngx_http_small_light_imagemagick.c [hunk 3]
```diff
@@ -337,21 +341,20 @@ ngx_int_t ngx_http_small_light_imagemagick_process(ngx_http_request_t *r, ngx_ht
 
         ngx_http_small_light_adjust_canvas_image_offset(&sz);
 
-        // check background fill
         backgroundfill_flg = ngx_http_small_light_parse_flag(NGX_HTTP_SMALL_LIGHT_PARAM_GET_LIT(&ctx->hash, "backgroundfill"));
```
Comment (kawakibi) [inline src/ngx_http_small_light_imagemagick.c]: line ini sampai satu block if `if (backgroundfill_flg == 1) {` gak dipakai, hapus aja

File: src/ngx_http_small_light_param.c [hunk 4]
```diff
@@ -95,6 +96,7 @@ static const ngx_str_t ngx_http_small_light_getparams[] = {
     ngx_string("arg_sharpen"),
     ngx_string("arg_unsharp"),
     ngx_string("arg_blur"),
+    ngx_string("arg_bluropt"),
```
Comment (kawakibi) [inline src/ngx_http_small_light_param.c]: full aja namingnya `bluroptimize` biar ga ambigu dikira blur option...

File: src/ngx_http_small_light_imagemagick.c [hunk 5]
```diff
@@ -384,60 +394,181 @@ ngx_int_t ngx_http_small_light_imagemagick_process(ngx_http_request_t *r, ngx_ht
...
+        DestroyPixelWand(bg_color);
+        DestroyMagickWand(merge_wand);
+    }
+
+    /* unsharp */
     unsharp = NGX_HTTP_SMALL_LIGHT_PARAM_GET_LIT(&ctx->hash, "unsharp");
     if (ngx_strlen(unsharp) > 0) {
         ParseGeometry(unsharp, &geo);
-        if (geo.rho > ctx->radius_max || geo.sigma > ctx->sigma_max) {
+        if (bluropt_flag != 0) {
```
Comment (kawakibi) [inline src/ngx_http_small_light_imagemagick.c]: pindah aja ini ke block if atas nya amcam `if (ngx_strlen(unsharp) > 0 && bluropt_flag == 0) {`

File: src/ngx_http_small_light_imagemagick.c [hunk 6]
```diff
@@ -384,60 +394,181 @@ ngx_int_t ngx_http_small_light_imagemagick_process(ngx_http_request_t *r, ngx_ht
...
             }
         }
     }
 
+    /* sharpen. */
     sharpen = NGX_HTTP_SMALL_LIGHT_PARAM_GET_LIT(&ctx->hash, "sharpen");
     if (ngx_strlen(sharpen) > 0) {
         ParseGeometry(sharpen, &geo);
-        if (geo.rho > ctx->radius_max || geo.sigma > ctx->sigma_max) {
+        if (bluropt_flag != 0) {
```
Comment (kawakibi) [inline src/ngx_http_small_light_imagemagick.c]: sama kek diatas

File: src/ngx_http_small_light_imagemagick.c [hunk 7]
```diff
@@ -384,60 +394,181 @@ ngx_int_t ngx_http_small_light_imagemagick_process(ngx_http_request_t *r, ngx_ht
...
             }
         }
     }
 
+    /* blur. */
     blur = NGX_HTTP_SMALL_LIGHT_PARAM_GET_LIT(&ctx->hash, "blur");
     if (ngx_strlen(blur) > 0) {
         ParseGeometry(blur, &geo);
-        if (geo.rho > ctx->radius_max || geo.sigma > ctx->sigma_max) {
+        if (bluropt_flag != 0) {
```
Comment (kawakibi) [inline src/ngx_http_small_light_imagemagick.c]: sama kek diatas

File: t/ngx_base/etc/basic.nginx.conf [hunk 8]
```diff
@@ -13,7 +13,10 @@ http {
         root www;
         small_light on;
         small_light_pattern_define too_big_image of=jpeg,dw=1000,dh=1000,da=l,q=95,e=imagemagick,jpeghint=y;
+        small_light_pattern_define too_big_image_avif of=avif,dw=1000,dh=1000,da=l,q=95,e=imagemagick,jpeghint=y;
         small_light_pattern_define ssize of=jpeg,dw=20,dh=20,da=l,q=95,e=imagemagick,jpeghint=y;
+        small_light_pattern_define ssize_webp of=webp,dw=20,dh=20,da=l,q=95,e=imagemagick,jpeghint=y;
+        small_light_pattern_define ssize_avif of=avif,dw=20,dh=20,da=l,q=95,e=imagemagick,jpeghint=y;
```
Comment (kawakibi) [inline t/ngx_base/etc/basic.nginx.conf:19]: tambahin buat bluroptimize ya

File: src/ngx_http_small_light_imagemagick.c (hunk 1 above)
Comment (AndhikaRei) [reply src/ngx_http_small_light_imagemagick.c]: Done

File: src/ngx_http_small_light_imagemagick.c (hunk 2 above)
Comment (AndhikaRei) [reply src/ngx_http_small_light_imagemagick.c]: Done

File: src/ngx_http_small_light_imagemagick.c (hunk 3 above)
Comment (AndhikaRei) [reply src/ngx_http_small_light_imagemagick.c]: Done

File: src/ngx_http_small_light_param.c (hunk 4 above)
Comment (AndhikaRei) [reply src/ngx_http_small_light_param.c]: Done

File: src/ngx_http_small_light_imagemagick.c (hunk 5 above)
Comment (AndhikaRei) [reply src/ngx_http_small_light_imagemagick.c]: Done

File: src/ngx_http_small_light_imagemagick.c (hunk 6 above)
Comment (AndhikaRei) [reply src/ngx_http_small_light_imagemagick.c]: Done

File: src/ngx_http_small_light_imagemagick.c (hunk 7 above)
Comment (AndhikaRei) [reply src/ngx_http_small_light_imagemagick.c]: Done

File: t/ngx_base/etc/basic.nginx.conf (hunk 8 above)
Comment (AndhikaRei) [reply t/ngx_base/etc/basic.nginx.conf:19]: Done

**Pull request merged**